add support for configuration via environment variables.

UiP9AV6Y opened the pull request:
the server instructions and authentication principals can be
configured via environment variables. this allows for an easier
deployment via docker/kubernetes, as you no longer need to mount
a configuration volume.
the configuration variables overwrite and settings from the CLI
if defined.

UiP9AV6Y committed:
add support for configuration via environment variables.

the server instructions and authentication principals can be
configured via environment variables. this allows for an easier
deployment via docker/kubernetes, as you no longer need to mount
a configuration volume.
the configuration variables overwrite and settings from the CLI
if defined.

angelbarrera92 commented:
Thanks for your contribution. I will take a look later! Awesome feature!

angelbarrera92 commented:
Hi @UiP9AV6Y 

With your change, it's required to have an `authn.yml` file even with the needed environment variables configured. 

I think its quite confusing. I suggest to rely only in one option, file or environment variable. 
File should takes precedence (make it blackguards compatible) to environment variable.

Thanks!

UiP9AV6Y committed:
add usage example using mounts/cli and environment variables.

two containers acting as reverse proxy for the same (debug) app.
one is configured via commandline arguments and volume mounts,
the other is entirely configured via environment variables.

UiP9AV6Y commented:
> With your change, it's required to have an authn.yml file even with the needed environment variables configured.

the *assets/authn.yaml* is already part of the docker image, which has by default no users configured.

you *can* use both cli/mounts and environment variables, but in that case you hopefully know what you are doing. the default usage expects to use either of the deployment workflows.

i have added an example to illustrate my point.
